etw/eventmetadata.go

```diff
@@ -0,0 +1,51 @@
+package etw
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+// InType indicates the type of data contained in the ETW event.
+type InType byte
+
+// Various InType definitions for tracelogging.
+const (
+	InTypeNull InType = iota

```

Comment:
Are these made up? Can we use any types here or do we have to use exactly these types?

Reply:
These have to match the tracelogging definitions (can be found in TraceLoggingProvider.h), so that decoding properly works. There are more than I put here, I just threw some in as a starting point.

Reply:
Can you write that in the comment. `These must match the values found in TraceLoggingProvider.h in the Windows SDK`. or something?